app/src/main/java/zapsolutions/zap/HomeActivity.java

```diff
@@ -406,4 +428,74 @@ public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, Strin
             }
         }
     }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+
+        PendingIntent pendingIntent = PendingIntent.getActivity(
+                this, 0, new Intent(this, getClass()).addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP), 0);
+        if (mNfcAdapter != null) {
+            mNfcAdapter.enableForegroundDispatch(this, pendingIntent, NfcUtil.IntentFilters(), null);
+        }
+    }
+
+    @Override
+    protected void onPause() {
+        super.onPause();
+        if (mNfcAdapter != null) {
+            mNfcAdapter.disableForegroundDispatch(this);
+        }
+    }
+
+    @Override
+    protected void onNewIntent(Intent intent) {
+        super.onNewIntent(intent);
+        NfcUtil.readTag(this, intent, new NfcUtil.OnNfcResponseListener() {
+            @Override
+            public void onSuccess(String payload) {
+                readInvoice(payload);
+            }
+        });
+    }
+
+    private void readInvoice(String invoice) {
+        InvoiceUtil.readInvoice(HomeActivity.this, compositeDisposable, invoice, new InvoiceUtil.OnReadInvoiceCompletedListener() {
+            @Override
+            public void onValidLightningInvoice(PayReq paymentRequest, String invoice) {
+                Intent intent = new Intent();
```

Comment:
Those intent creations for bitcoin/lightning could be instance methods of SendBsdFragment
like: SendBSDFragment.createLightning(lnpayrequest, lninvoice) and SendBSDFragment.createOnChain(address,amount,message)

Reply:
done